File: frontend/src/api/announcement.ts
/**
 * 系统公告 API
 */
import request from './index'

export type AnnouncementPriority = 'normal' | 'important' | 'urgent'

export interface Announcement {
  id: number
  title: string
  content: string
  priority: AnnouncementPriority
  is_active: boolean
  author_id: number
  author_name?: string
  created_at: string
  updated_at: string
}

export interface AnnouncementListResponse {
  total: number
  items: Announcement[]
}

export interface AnnouncementCreateParams {
  title: string
  content: string
  priority: AnnouncementPriority
  is_active: boolean
}

export interface AnnouncementUpdateParams {
  title?: string
  content?: string
  priority?: AnnouncementPriority
  is_active?: boolean
}

/** 获取公告列表（所有用户）*/
export function getAnnouncements(includeInactive = false): Promise<AnnouncementListResponse> {
  return request.get('/api/v1/announcements', {
    params: includeInactive ? { include_inactive: true } : {},
  })
}

/** 创建公告（仅系统管理员）*/
export function createAnnouncement(data: AnnouncementCreateParams): Promise<Announcement> {
  return request.post('/api/v1/announcements', data)
}

/** 更新公告（仅系统管理员）*/
export function updateAnnouncement(id: number, data: AnnouncementUpdateParams): Promise<Announcement> {
  return request.put(`/api/v1/announcements/${id}`, data)
}

/** 删除公告（仅系统管理员）*/
export function deleteAnnouncement(id: number): Promise<void> {
  return request.delete(`/api/v1/announcements/${id}`)
}

File: backend/app/schemas/announcement.py
"""系统公告 Pydantic 模式"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class AnnouncementCreate(BaseModel):
    """创建公告请求"""
    title: str = Field(..., min_length=1, max_length=200, description="公告标题")
    content: str = Field(..., min_length=1, description="公告内容")
    priority: str = Field("normal", description="优先级: normal / important / urgent")
    is_active: bool = Field(True, description="是否启用")


class AnnouncementUpdate(BaseModel):
    """更新公告请求"""
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    content: Optional[str] = Field(None, min_length=1)
    priority: Optional[str] = None
    is_active: Optional[bool] = None


class AnnouncementResponse(BaseModel):
    """公告响应"""
    id: int
    title: str
    content: str
    priority: str
    is_active: bool
    author_id: int
    author_name: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AnnouncementListResponse(BaseModel):
    """公告列表响应"""
    total: int
    items: List[AnnouncementResponse]
